internal/addon/authentication/provider.go

```diff
@@ -107,14 +100,19 @@ func (sp *secretsProvider) GenerateSecrets(targetAuthType map[string]string) (ma
 		}
 	}
 
+	err := sp.injectCA(targetAuthType, secretKeys)
+	if err != nil {
+		return nil, err
+	}
+
 	return secretKeys, nil
 }
 
 func (sp *secretsProvider) FetchSecrets(targetsSecret map[string]client.ObjectKey, targetAnnotation string) ([]corev1.Secret, error) {
```

Comment:
I see now a bunch of pieces of code that are using a `map[string]something` and I miss the typed string like we have the `addon.Signal` in the `NewSecretsProvider` function. This helps a lot in understand what's supposed to be in this map, given the exported function lacks a doc-comment.